fineract-provider/src/main/java/org/apache/fineract/notification/domain/NotificationMapperRepository.java

```diff
@@ -18,6 +18,25 @@
  */
 package org.apache.fineract.notification.domain;
 
+import java.util.Collection;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Modifying;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import org.springframework.transaction.annotation.Transactional;
 
-public interface NotificationMapperRepository extends JpaRepository<NotificationMapper, Long> {}
+public interface NotificationMapperRepository extends JpaRepository<NotificationMapper, Long> {
+
+    @Transactional
```

Comment:
No need for transactional here. Tx management should be around a business functionality, not a repository call.

Reply:
When I leave out Transactional (from org.springframework.transaction.annotation.Transactional) I get the TransactionRequiredException on calling any of the methods where I have used.